packages/evm/test/payments/payments-distribution.js

```diff
@@ -0,0 +1,437 @@
+          })
+
+          if (governorShare.eq(bn(0))) {
+            it('ignores governor fees request', async () => {
+              const previousTokenBalance = await token.balanceOf(paymentsBook.address)
+              const previousAnotherTokenBalance = await token.balanceOf(paymentsBook.address)
+              const previousEthBalance = bn(await web3.eth.getBalance(paymentsBook.address))
+
+              await paymentsBook.transferManyGovernorFees(periodId, [token.address, anotherToken.address, eth.address])
+
+              const currentTokenBalance = await token.balanceOf(paymentsBook.address)
+              assertBn(currentTokenBalance, previousTokenBalance, 'payments book token balance does not match')
+
+              const currentAnotherTokenBalance = await token.balanceOf(paymentsBook.address)
+              assertBn(currentAnotherTokenBalance, previousAnotherTokenBalance, 'payments book another token balance does not match')
+
+              const currentEthBalance = bn(await web3.eth.getBalance(paymentsBook.address))
+              assertBn(currentEthBalance, previousEthBalance, 'payments book eth balance does not match')
+            })
+          } else {
+            it('estimates governor fees correctly', async () => {
+              const fees = await paymentsBook.getGovernorFees(periodId, token.address)
+              const otherFees = await paymentsBook.getManyGovernorFees(periodId, [anotherToken.address, eth.address])
+
+              assertBn(fees, expectedGovernorTokenFees, 'governor token fees does not match')
+              assertBn(otherFees[0], expectedGovernorAnotherTokenFees, 'governor another token fees does not match')
+              assertBn(otherFees[1], expectedGovernorEthFees, 'governor eth fees does not match')
+            })
+
+            it('transfers fees to the governor', async () => {
+              const previousBalance = await token.balanceOf(governor)
+
+              await paymentsBook.transferGovernorFees(periodId, token.address)
+
+              const fees = await paymentsBook.getGovernorFees(periodId, token.address)
+              assertBn(fees, 0, 'governor token fees does not match')
+
+              const currentBalance = await token.balanceOf(governor)
+              assertBn(currentBalance, previousBalance.add(expectedGovernorTokenFees), 'governor token balance does not match')
+            })
+
+            it('ignores duplicated governor requests', async () => {
+              const previousBalance = await token.balanceOf(governor)
+
+              await paymentsBook.transferGovernorFees(periodId, token.address)
+              await paymentsBook.transferGovernorFees(periodId, token.address)
+
+              const currentBalance = await token.balanceOf(governor)
+              assertBn(currentBalance, previousBalance.add(expectedGovernorTokenFees), 'governor token balance does not match')
+            })
```

Comment:
Might be worth double checking that the second call doesn't emit any events

Reply:
Addressed in b3b4648025045022dd2b426518dc3c6f756bc195